tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLIterationTest.java

```diff
@@ -54,14 +54,64 @@ public void setup() {
 
     @Test
     public void testNegative() {
-        Assert.assertEquals(negative.getErrorCount(), 2);
+        Assert.assertEquals(negative.getErrorCount(), 17);
         int index = 0;
         BAssertUtil.validateError(negative, index++,
-                                  "invalid tuple binding pattern: expected a tuple type, but found '(xml|string)'",
-                                  11, 17);
+                                  "invalid tuple binding pattern: attempted to infer a tuple type, but found 'xml'",
+                                  13, 17);
         BAssertUtil.validateError(negative, index++, "incompatible types: expected " +
                 "'function ((xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text)) returns ()'," +
-                " found 'function ([int,xml,string]) returns ()'", 16, 19);
+                " found 'function ([int,xml,string]) returns ()'", 18, 19);
+        BAssertUtil.validateError(negative, index++,
+                "incompatible types: expected 'other', found 'xml:Element'",
```

Comment:
Ideally, this error shouldn't be logged. The actual error is the next one, `'xml:Element' is not an iterable collection`.